### PR TITLE
Update translations for 0.14.0

### PR DESCRIPTION
**Pull request by rmol**
## Status

Ready for review

## Description of Changes

New translations for the 0.14.0 release. 

Also includes small changes to `Makefile` and `securedrop/i18n_tool.py` to ensure that `make test-config` results in a `securedrop/config.py` in which `SUPPORTED_LOCALES` contains an entry for every supported language.

## Testing

For each supported language, check that each of the 18 source strings changed for 0.14.0 has a plausible-looking translation.

Check out this branch, then run `securedrop/bin/dev-shell ./i18n_tool.py --verbose translate-messages --compile` to create the `.mo` files.

Then save this script as `check_translations.py`:
```
#!/usr/bin/env python3

import gettext
import os
import sys


SUPPORTED_LANGUAGES = [
    "ar",
    "ca",
    "de_DE",
    "el",
    "es_ES",
    "en_US",
    "fr_FR",
    "hi",
    "it_IT",
    "nb_NO",
    "nl",
    "pt_BR",
    "ro",
    "ru",
    "sv",
    "tr",
    "zh_Hant",
]

TESTS = [
    "<a href=\"/\">Refresh this page</a>, and you're done!",
    "Change Name &amp; Admin Status",
    "Change Name",
    "Click the <img src=\"{icon}\" alt=\"shield icon\"> in the toolbar above",
    "Field can not be more than {max_chars} characters.",
    "First name and last name are optional",
    "First name",
    "Last name",
    "Name not updated: {}",
    "Name updated.",
    "RESET APP TOKEN",
    "RESET HARDWARE TOKEN",
    "Reset 2FA for hardware tokens like Yubikey",
    "Reset 2FA for mobile apps such as FreeOTP or Google Authenticator",
    "SecureDrop uses automatically generated diceware passwords.",
    "Select <strong>Advanced Security Settings</strong>",
    "Select <strong>Safest</strong>",
    (
        "Your Tor Browser's <a id=\"disable-js\" href=\"\"><strong>Security Setting</"
        "strong></a> is too low. Use the <img src=\"{icon}\" alt=\"shield icon\"> "
        "&nbsp;button in your browser’s toolbar to change it."
    ),
]


if __name__ == "__main__":
    translation_dir = sys.argv[1]

    for lang in SUPPORTED_LANGUAGES:
        print("Checking {}...".format(lang))
        os.environ["LANGUAGE"] = lang
        gettext.bindtextdomain("messages", translation_dir)
        missing = False
        for i, t in enumerate(TESTS, 1):
            translation = gettext.gettext(t)
            print("  {}: {}".format(i, translation))
            if lang != "en_US" and translation == t:
                missing = True
                print('MISSING TRANSLATION FOR "{}"'.format(t), file=sys.stderr)

        if not missing:
            print("Looks good.")
```
When run with the translations directory as its only argument (e.g. `check_translations.py securedrop/translations`, from the root of your working copy), you should see the 18 strings translated for each supported language, with no reports of missing translations, and no unexpected markup (that doesn't match the source string) or other suspicious content in the translated strings.

Additionally:

- [ ] Ensure there is a passing CI translation job (ran on all pages-layout tests) - you need to do this by triggering a CI run from the shared remote.
- [ ] Review the diff manually to verify there are no unusual or unexpected changes in the diff. Note that Catalan is entirely new this release, and there have been translation updates other than the 18 source string changes we made for 0.14.0. It's a big diff, but you can get through it faster (assuming you're using `less` as your pager) by searching with this regular expression: `\+("|msgid|msgstr)`.

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make -C securedrop test`) pass in the development container

### CODE REVIEW

**Comment by kushaldas** [comment]
Except that one failed test, rest looks good. I am still trying to debug on that error on `self._journalist_delete_one()` for `de_DE`.

**Comment by kushaldas** [comment]
Also, a separate point for future, locally most of the time, the 3 tests related to `htop/totp` are also failing for me.

**Comment by kushaldas** [comment]
I can see the same happening in `es_ES` too, if I manually execute the test. The following is the screenshot just before `el.click()` call.

why_are_you_breaking

**Comment by kushaldas** [comment]
Below is the screenshot for `ar` language where the same test is passing.

wtf-is-this-working then

**Comment by kushaldas** [comment]
I did a string review of the translation files, they look good to me.

File: install_files/ansible-base/roles/tails-config/templates/nl.po [hunk 1]
```diff
@@ -10,7 +10,7 @@ msgstr ""
 "PO-Revision-Date: 2017-12-01 13:48+0000\n"
 "Last-Translator: kwadronaut <kwadronaut@autistici.org>\n"
 "Language-Team: Dutch "
-"<https://weblate.securedrop.org/projects/securedrop/desktop/nl/>\n"
+"<https://weblate.securedrop.club/projects/securedrop/desktop/nl/>\n"
```
Comment (redshiftzero) [inline install_files/ansible-base/roles/tails-config/templates/nl.po]: .club is back?

**Pull request merged**